simple.lua

```diff
@@ -0,0 +1,20 @@
+-- Written by [https://github.com/catwell](catwell) and modified by DarkWarrior703
+local counts = {}
+local t = {}
+
+local function insert(s)
+    counts[s] = (counts[s] or 0) + 1
+end
+-- Read the whole buffer, lower it and then create the count table
+string.gsub(io.read("*all"):lower(), "(%S+)", insert)
```

Comment:
I'm guessing this is reading the entire input file into memory? In which case, that goes against the constraints I've set in the article. You need to read line by line (`read("*line")`) -- hopefully Lua buffers file I/O by default behind the scenes.